src/geyser/bin.zig

```diff
@@ -0,0 +1,278 @@
+        const owner_pubkey = try sig.core.Pubkey.fromString(owner_str);
+        owner_pubkeys.putAssumeCapacity(owner_pubkey, {});
+    }
+
+    return owner_pubkeys;
+}
+
+pub fn csvDump() !void {
+    const allocator = std.heap.c_allocator;
+
+    const metrics_thread = try sig.prometheus.spawnMetrics(allocator, 12355);
+    metrics_thread.detach();
+    std.debug.print("spawing metrics thread on port 12355\n", .{});
+
+    const pipe_path = config.pipe_path;
+    std.debug.print("using pipe path: {s}\n", .{pipe_path});
+
+    // owner filters
+    var maybe_owner_pubkeys = try getOwnerFilters(allocator);
+    defer if (maybe_owner_pubkeys) |*owners| owners.deinit();
+    if (maybe_owner_pubkeys) |owner_pubkeys| {
+        std.debug.print("owner filters: {s}\n", .{owner_pubkeys.keys()});
+    } else {
+        std.debug.print("owner filters: none\n", .{});
+    }
+
+    // csv file to dump to
+    const dump_csv_path = try std.fmt.allocPrint(
+        allocator,
+        "{s}{s}",
+        .{ sig.VALIDATOR_DIR, "accounts.csv" },
+    );
+    defer allocator.free(dump_csv_path);
+
+    const csv_file = try std.fs.cwd().createFile(dump_csv_path, .{});
+    defer csv_file.close();
+
+    std.debug.print("dumping to csv: {s}\n", .{dump_csv_path});
+
+    // setup reader
+    const exit = try allocator.create(std.atomic.Value(bool));
+    defer allocator.destroy(exit);
+    exit.* = std.atomic.Value(bool).init(false);
+
+    var reader = try sig.geyser.GeyserReader.init(allocator, pipe_path, exit, .{
+        .bincode_buf_len = config.geyser_bincode_buf_len,
+        .io_buf_len = config.geyser_io_buf_len,
+    });
+    defer reader.deinit();
+
+    // preallocate memory for csv rows
+    const recycle_fba = try allocator.create(sig.utils.allocators.RecycleFBA(.{ .thread_safe = true }));
+    recycle_fba.* = try sig.utils.allocators.RecycleFBA(.{ .thread_safe = true }).init(allocator, 1 << 32);
+    defer {
+        recycle_fba.deinit();
+        allocator.destroy(recycle_fba);
+    }
+
+    // setup thread to write to csv
+    var io_channel = try sig.sync.Channel([]u8).create(allocator);
+    defer {
+        io_channel.deinit();
+        allocator.destroy(io_channel);
+    }
+
+    const io_handle = try std.Thread.spawn(.{}, csvDumpIOWriter, .{ exit, csv_file, io_channel, recycle_fba });
+    defer io_handle.join();
+    errdefer {
+        exit.store(true, .release);
+    }
+
+    // start to read from geyser
+    while (true) {
+        _, const payload = try reader.readPayload();
+        defer reader.resetMemory();
+
+        if (std.meta.activeTag(payload) == .EndOfSnapshotLoading) {
+            break;
+        }
+
+        switch (std.meta.activeTag(payload)) {
+            .AccountPayloadV1 => {},
+            .EndOfSnapshotLoading => {
+                std.debug.print("recv end of snapshot loading signal\n", .{});
+                exit.store(true, .release);
+                break;
+            },
+        }
```

Comment:
it looks like the first 3 lines are extra.

just curious, why do you stop reading from Geyser at the end of the snapshot?

In my understanding, this event indicates switching the `is_startup` flag, on which relies the `update_account` function, from `true` to `false`.
I believe it's better to leave a comment to describe what should be implemented in the next step (e.g buffering and dumping of ongoing data)

Reply:
[0775da7](https://github.com/Syndica/sig/pull/279/commits/0775da7ac8a41c3bb16610db0544818e3b1e5764)